### PR TITLE
[CFM_126] chore: Create settings for unit testing

### PR DESCRIPTION
**Pull request by seanred360**
This sets up unit testing with Jest and React Testing Library.
Use the command "yarn test' to run all tests in the project.
Tests for page components must go in the __test__ directory.
Tests should go in the same directory as the component it is testing.

A Github action to Build, Test, and Lint pull requests is also included.

github-actions

### CODE REVIEW

**Review by jaronaearle — CHANGES_REQUESTED** [review]
I would suggest adding a simple test or two to some of the components to actually verify that this is config is set up properly.

**Comment by seanred360** [comment]
> I would suggest adding a simple test or two to some of the components to actually verify that this is config is set up properly.

Thanks for taking a look! I actually have some tests I just was not sure if I should include them or not.

**Comment by seanred360** [comment]
> I would suggest adding a simple test or two to some of the components to actually verify that this is config is set up properly.

I added a few very simple tests. The one that tests format.js fails because the function doesn't work. I have a fix in another PR that will pass the test. Currently I am trying to figure out how to test components. Most of the components we use are abstracted behind MUI and have state and other dependencies that are hard to insolate.

**Comment by jaronaearle** [comment]
> > I would suggest adding a simple test or two to some of the components to actually verify that this is config is set up properly.
> 
> I added a few very simple tests. The one that tests format.js fails because the function doesn't work. I have a fix in another PR that will pass the test. Currently I am trying to figure out how to test components. Most of the components we use are abstracted behind MUI and have state and other dependencies that are hard to insolate.

If you'd like some help getting started with testing components let me know. I should have some time tomorrow or over the weekend and we could pair on that if you'd like.

**Comment by seanred360** [comment]
> > > I would suggest adding a simple test or two to some of the components to actually verify that this is config is set up properly.
> > 
> > 
> > I added a few very simple tests. The one that tests format.js fails because the function doesn't work. I have a fix in another PR that will pass the test. Currently I am trying to figure out how to test components. Most of the components we use are abstracted behind MUI and have state and other dependencies that are hard to insolate.
> 
> If you'd like some help getting started with testing components let me know. I should have some time tomorrow or over the weekend and we could pair on that if you'd like.

I would love that! I'll Discord you.

File: package-lock.json
Comment (doomjar) [inline package-lock.json:1]: This lock file should be deleted as we should only be using one package manager. After removing this file, run `yarn install` and add/commit the yarn.lock file.

Comment (seanred360) [reply package-lock.json:1]: I fixed the issue thanks!

**Review by jaronaearle — DISMISSED** [review]
Solid work, lgtm 🚀

**Review by bernardm — APPROVED** [review]

**Review by bernardm — APPROVED** [review]

**Pull request merged**